### PR TITLE
Feat/testing approaches

### PR DESCRIPTION
**Pull request by G0maa**
- E2E, Integration (kind of) & unit testing (also kind of).

### CODE REVIEW

File: apps/api/test/TestManager.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                resave: false,
+                saveUninitialized: false,
+                cookie: {
+                    maxAge: configService.getOrThrow<number>("COOKIE_MAX_AGE"),
+                    httpOnly: true,
+                },
+                store: new RedisStore({
+                    client: redisClient,
+                }),
+            }),
```
[the remark this answers is not in the document]
Comment (G0maa) [reply apps/api/test/TestManager.ts:47]: @abdo0nasser

File: apps/api/schema.graphql
Comment (bassiounix) [inline apps/api/schema.graphql:1]: Why are that fields like `created_at` nullable?

Comment (G0maa) [reply apps/api/schema.graphql:1]: @bassiounix iirc this is the type of the result of `find` by Drizzle.

File: apps/api/test/setup.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+export default async (): Promise<void> => {
+    console.log("# Started Jest globalSetup.");
+    const testManager = new TestManager();
+
+    await testManager.beforeAll();
+
+    await testManager.app.init();
+
+    // TODO: Apply Database migrations/seeders.
```
Comment (bassiounix) [inline apps/api/test/setup.ts:14]: Where are the db stuff?

File: apps/api/test/TestManager.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import RedisStore from "connect-redis";
+import * as session from "express-session";
+import * as passport from "passport";
+import { createClient } from "redis";
+import { AppModule } from "../src/app.module";
+
+export class TestManager {
+    // biome-ignore lint/suspicious/noExplicitAny: it is any.
+    public httpServer: any;
+    public app: INestApplication;
```
Comment (bassiounix) [inline apps/api/test/TestManager.ts:13]: `INestApplication<NestExpressApplication>` could be more expressive. not mandatory.

File: apps/api/src/app.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { ContextIdFactory } from "@nestjs/core";
```
Comment (bassiounix) [inline apps/api/src/app.service.spec.ts:1]: Unused imports!!!

File: apps/api/package.json
Comment (bassiounix) [inline apps/api/package.json:1]:
```txt
A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks. Active timers can also cause this, ensure that .unref() was called on them.
```
appears when trying to run `jest`.

Also `test:e2e` shows a similar message:
```txt
Force exiting Jest: Have you considered using `--detectOpenHandles` to detect async operations that kept running after all tests finished?
```

**Review by bassiounix — COMMENTED** [review]
Test coverage is low, not many functions are being tested.
Otherwise .. LGTM.

File: apps/api/test/setup.ts (hunk 2 above)
Comment (G0maa) [reply apps/api/test/setup.ts:14]: Hmm... I can't recall why I didn't add them, I can work on them this weekend, but I prefer that we merge this anway.

File: apps/api/package.json
Comment (G0maa) [reply apps/api/package.json:1]: @bassiounix 
The intention of this branch is not to test. The intention is to show developer working on Disworse HOW to test.

Comment (G0maa) [reply apps/api/package.json:1]: Also on the Jest error, I never managed to fix it, I usually silence it with `--forceExit`.

File: apps/api/test/setup.ts (hunk 2 above)
Comment (G0maa) [reply apps/api/test/setup.ts:14]: I spent about ~1 hour trying to figure this out. Two issues created regarding this.
https://github.com/activecourses/disworse/issues/76
https://github.com/activecourses/disworse/issues/77

**Pull request merged**